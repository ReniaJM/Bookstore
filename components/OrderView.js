import React, {Component} from 'react'

class OrderView extends Component {
    render(){
        return(
            <div className='orderView row'>
                <div className='col-md-8'>
                    <span>{this.props.book.name}</span>
                </div>
                <div className='col-md-4'>
                    <button className='btn btn-danger' onClick={(event) =>this.props.removeFromOrder(this.props.book.name)}>usu≈Ñ</button>
                </div>
            </div>
        )
    }
}

export default OrderView