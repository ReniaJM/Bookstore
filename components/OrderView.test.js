import React from 'react';
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import OrderView from "./OrderView";

configure({adapter: new Adapter()})
describe('order test', () =>{
    it('snapshot matches', () =>{
        const book= {
            name: 'Book title'
        }
        const wrapper =shallow (<OrderView  book={book}/>)
        expect(wrapper).toMatchSnapshot();
    })

    it('title of the book is displayed as in props', () =>{
        const book= {
            name: 'Book title'
        }
        const wrapper =shallow (<OrderView  book={book}/>)
        expect(wrapper.find('.orderView').find('span').text()).toBe(book.name)
    })
})