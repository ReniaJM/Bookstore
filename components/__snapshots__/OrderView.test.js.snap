// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`order test snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderView
    book={
      Object {
        "name": "Book title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <b>
          Book title
        </b>,
        <br />,
        <button
          className="btn btn-danger"
          onClick={[Function]}
        >
          usuń
        </button>,
      ],
      "className": "orderView",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Book title",
        },
        "ref": null,
        "rendered": "Book title",
        "type": "b",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "usuń",
          "className": "btn btn-danger",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "usuń",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <b>
            Book title
          </b>,
          <br />,
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            usuń
          </button>,
        ],
        "className": "orderView",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Book title",
          },
          "ref": null,
          "rendered": "Book title",
          "type": "b",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "usuń",
            "className": "btn btn-danger",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "usuń",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`order test title of the book is displayed as in props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderView
    book={
      Object {
        "name": "Book title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <b />,
        <br />,
        <button
          className="btn btn-danger"
          onClick={[Function]}
        >
          usuń
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "b",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "usuń",
          "className": "btn btn-danger",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "usuń",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <b />,
          <br />,
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            usuń
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "b",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "usuń",
            "className": "btn btn-danger",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "usuń",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
